### PR TITLE
Nested dependencies testing

### PR DESCRIPTION
**Pull request by am-houtkooper**
This PR adds support for testing nested dependencies using `@SUT`, `@MC` and `@SC`.
If Service `A` depends on `B` which in turn depends on `C`. Then this PR allows you to test `A` using `B` while mocking `C`. This is necessary for doing integration testing where a low-level system access service needs to be mocked only.

So now we can declare `@SUT A a` and `@MC C c` which instantiates the intermediate service `B`.

Example:
`org/jvnet/testing/hk2mockito/IndirectCollaboratorMockInjectionTest.java`

At the same time it provides a `JUnitRunner` interface to avoid having to use inheritance. It provides a way to write JUnit tests that conform with the way tests are written in JUnit. It loses the more fine-grained control that comes with inheritance.

### CODE REVIEW

File: hk2-testing/hk2-mockito/src/test/java/org/jvnet/testing/hk2mockito/MultipleInjectionTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package org.jvnet.testing.hk2mockito;
```
Comment (Cousjava) [inline hk2-testing/hk2-mockito/src/test/java/org/jvnet/testing/hk2mockito/MultipleInjectionTest.java]: Copyright notice needs adding

File: hk2-testing/hk2-mockito/src/test/java/org/jvnet/testing/hk2mockito/IndirectCollaboratorMockInjectionTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2014, 2018 Oracle and/or its affiliates. All rights reserved.
```
Comment (Cousjava) [inline hk2-testing/hk2-mockito/src/test/java/org/jvnet/testing/hk2mockito/IndirectCollaboratorMockInjectionTest.java]: If you don't work for Oracle then this should be yourself instead, unless this file is a rename from elsewhere that git hasn't picked up on.

File: hk2-testing/hk2-mockito/src/test/java/org/jvnet/testing/hk2mockito/fixture/service/MultipleConstructorInjectionService.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package org.jvnet.testing.hk2mockito.fixture.service;
```
Comment (Cousjava) [inline hk2-testing/hk2-mockito/src/test/java/org/jvnet/testing/hk2mockito/fixture/service/MultipleConstructorInjectionService.java]: Copyright notice needs adding

**Comment by am-houtkooper** [comment]
> The code itself looks fine, just a couple of copyright notices that need adding.

Thanks! Resolved the copyright notices.

**Pull request merged**